lib/Cake/Utility/Folder.php

```diff
@@ -399,21 +399,26 @@ public function inCakePath($path = '') {
 	}
 
 /**
- * Returns true if the File is in given path.
+ * Returns true if the Folder is in the given path.
  *
- * @param string $path The path to check that the current pwd() resides with in.
- * @param bool $reverse Reverse the search, check that pwd() resides within $path.
+ * @param string $path The absolute path to check that the current `pwd()` resides within.
+ * @param bool $reverse Reverse the search, check if the given `$path` resides within the current `pwd()`.
  * @return bool
+ * @throws \InvalidArgumentException When the given `$path` argument is not an absolute path.
  * @link http://book.cakephp.org/2.0/en/core-utility-libraries/file-folder.html#Folder::inPath
  */
 	public function inPath($path = '', $reverse = false) {
+		if (!Folder::isAbsolute($path)) {
+			throw new InvalidArgumentException(__d('cake_dev', 'The $path argument is expected to be an absolute path.'));
```

Comment:
Do we need to intoduce the cake_dev translation here?
IMO not needed for new ones.